src/Microsoft.OpenApi/Models/OpenApiError.cs

```diff
@@ -3,13 +3,36 @@
 
 using Microsoft.OpenApi.Exceptions;
 
-namespace Microsoft.OpenApi.Readers
+namespace Microsoft.OpenApi.Models
 {
     /// <summary>
-    /// Error related to reading Open API YAML/JSON.
+    /// Error reason.
+    /// </summary>
+    public enum ErrorReason
```

Comment:
>ErrorReas [](start = 16, length = 9)

Is this needed asking b/c from a consumer perspective i am not able to figure out what do these mean? 

I looked at the exceptions themselves and they are descriptive enough to convey the problem e.g. EmailMustBeEmailFormat

Reply:
It is possible that now we are including the RuleName in the error, the reason is less useful.  However, the reason does allow some amount of statistical analysis on documents to understand the types of errors that people are making.  Arguably this information could be derived from well known rule names.  I'm on the fence.  Perhaps Sam has thoughts.  /cc @xuzhg

Reply:
Oh ok, i do not have strong opinion on removing it but feel it should be little bit more descriptive like IncorrectFormat and RequiredFieldMissing or something along those lines?

---
In reply to: [178636972](https://github.com/Microsoft/OpenAPI.NET/pull/235#discussion_r178636972) [](ancestors = 178636972)

Reply:
I agree with this part: "Arguably this information could be derived from well known rule names."

Having to specify this ErrorReason together with the rule name seems redundant, and as Shweta mentioned, it's a little difficult to understand what these reasons mean. I'd say we keep the statistical analysis concern separate --that can be done by postprocessing the rule name.

---
In reply to: [178638357](https://github.com/Microsoft/OpenAPI.NET/pull/235#discussion_r178638357) [](ancestors = 178638357,178636972)

Reply:
If we want to facilitate any analysis later, maybe we can build a map containing information about a rule  (rule -> ruleInformation) with ruleInformation containing things like these enums? Users can add info to that map for their custom rules.

---
In reply to: [178909044](https://github.com/Microsoft/OpenAPI.NET/pull/235#discussion_r178909044) [](ancestors = 178909044,178638357,178636972)